src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/DependencyInjection/ApiBehaviorOptionsSetup.cs

```diff
@@ -2,47 +2,19 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Collections.Generic;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.AspNetCore.Mvc.Core;
 using Microsoft.AspNetCore.Mvc.Infrastructure;
-using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 
 namespace Microsoft.Extensions.DependencyInjection
 {
-    internal class ApiBehaviorOptionsSetup :
-        ConfigureCompatibilityOptions<ApiBehaviorOptions>,
-        IConfigureOptions<ApiBehaviorOptions>
+    internal class ApiBehaviorOptionsSetup : IConfigureOptions<ApiBehaviorOptions>, IPostConfigureOptions<ApiBehaviorOptions>
     {
         internal static readonly Func<ActionContext, IActionResult> DefaultFactory = DefaultInvalidModelStateResponse;
         internal static readonly Func<ActionContext, IActionResult> ProblemDetailsFactory = ProblemDetailsInvalidModelStateResponse;
 
-        public ApiBehaviorOptionsSetup(
-            ILoggerFactory loggerFactory,
-            IOptions<MvcCompatibilityOptions> compatibilityOptions)
-            : base(loggerFactory, compatibilityOptions)
-        {
-        }
-
-        protected override IReadOnlyDictionary<string, object> DefaultValues
-        {
-            get
-            {
-                var dictionary = new Dictionary<string, object>();
-
-                if (Version < CompatibilityVersion.Version_2_2)
```

Comment:
This was the only `<` check of a `CompatibilityVersion`. Everywhere else, just made the `xyzSetup` classes set the new (`Version_3_0`) defaults.